extensions/wikia/WallNotifications/WallNotifications.class.php

```diff
@@ -387,50 +387,65 @@ protected function sendEmails( array $watchers, WallNotificationEntity $notifica
 			$watcher = $this->getUser( $val );
 			$mode = $watcher->getOption( 'enotifwallthread' );
 
-			if ( !empty( $mode ) && $watcher->getId() != 0 && (
+			if ( !( $watcher->getBoolOption('unsubscribed') === true ) &&
+				!empty( $mode ) && $watcher->getId() != 0 && (
 				( $mode == WALL_EMAIL_EVERY ) ||
 				( $mode == WALL_EMAIL_SINCEVISITED && empty( $this->uniqueUsers[$entityKey][$watcher->getId()] ) )
 			)) {
 
-				$key = $this->createKeyForMailNotification( $watcher->getId(), $notification );
-				$watcherName = $watcher->getName();
-
-				if ( $notification->data->msg_author_username == $notification->data->msg_author_displayname ) {
-					$author_signature = $notification->data->msg_author_username;
+				$controller = $this->getEmailExtensionController( $notification );
```

Comment:
What're your thoughts on updating this method to return a boolean and be something like `isForumPost`? For me when reading this it's not really clear why we're getting the emailExtensionController, like what the intent is. When looking into `getEmailExtensionController()` then it's revealed that we care if this is a forum post or not. I think it may be clearer if we indicate that here, and then we could hardcode the forum controller on L409.